### PR TITLE
BG-9197: Add new /acceleratetx route to BitGo Express

### PR DESCRIPTION
**Pull request by paddyquinn**
This branch branches from BG-9197. We should merge #162 first and then I will rebase this to branch from master. Only `src/clientRoutes.js` needs review as the other file will be reviewed in #162. Sorry this is confusing. This probably should have all been one PR.

### CODE REVIEW

**Comment by tylerlevine** [comment]
Why are we exposing this feature as "bump"? Can we not reuse the v1 terms for some reason (accelerateTransaction)?

**Comment by paddyquinn** [comment]
> Why are we exposing this feature as "bump"? Can we not reuse the v1 term (accelerateTransaction) for some reason ?

That is just the naming that came out of our design meeting. Not sure it was deliberate. @OttoAllmendinger @Xekyo thoughts on using the v1 terminology "accelerate" vs "bump"?

File: src/clientRoutes.js [hunk 1]
```diff
@@ -572,6 +580,8 @@ exports = module.exports = function(app, args) {
 
   app.post('/api/v2/:coin/wallet/:id/sweep', parseBody, prepareBitGo(args), promiseWrapper(handleV2Sweep, args));
 
+  // CPFP
+  app.post('/api/v2/:coin/wallet/:id/bump', parseBody, prepareBitGo(args), promiseWrapper(handleV2BumpTransaction, args));
```
Comment (OttoAllmendinger) [inline src/clientRoutes.js]: with `accelerateTransaction` as the func name we should call this `acceleratetx` (see `signtx`)

Comment (paddyquinn) [reply src/clientRoutes.js]: done

**Comment by tylerlevine** [comment]
Closing and reopening to retrigger travis

File: src/clientRoutes.js [hunk 2]
```diff
@@ -342,6 +342,14 @@ const handleV2Sweep = co(function *handleV2Sweep(req) {
   return wallet.sweep(req.body);
 });
 
+// handle CPFP accelerate transaction creation
+const handleV2AccelerateTransaction = co(function *handleV2BumpTransaction(req) {
```
Comment (Xekyo) [inline src/clientRoutes.js]: Is the "handleV2BumpTransaction" still accurate? Shouldn't it be "…accelerateTx"?

File: src/v2/wallet.js [hunk 3]
```diff
@@ -1135,6 +1136,28 @@ Wallet.prototype.prebuildAndSignTransaction = function(params, callback) {
...
+  return co(function *() {
+    // TODO(BG-9349): change the last check to > 0 and the error message once platform allows multiple transactions to
+    //                be bumped in the same CPFP transaction
+    if (_.isUndefined(params.cpfpTxIds) || !Array.isArray(params.cpfpTxIds) || params.cpfpTxIds.length !== 1) {
+      const error = new Error('expecting cpfpTxIds to be an array of length 1');
+      error.code = 'cpfptxids_not_array';
+      throw error;
+    }
+
+    if (_.isUndefined(params.cpfpFeeRate) && _.isUndefined(params.maxFeeRate)) {
```
Comment (Xekyo) [inline src/v2/wallet.js]: I don't know if it's usually done at this stage, but should we perhaps check the input type for these?

**Comment by paddyquinn** [comment]
Closing this for now, until #162 is merged since this is based on that branch.

**Comment by paddyquinn** [comment]
Replaced by #170

File: src/clientRoutes.js (hunk 2 above)
Comment (paddyquinn) [reply src/clientRoutes.js]: Done. See #170

File: src/v2/wallet.js (hunk 3 above)
Comment (paddyquinn) [reply src/v2/wallet.js]: Resolved in #162

**Pull request closed without merging**